### PR TITLE
Fix windows remote attach experience

### LINKED ISSUE #3758: Windows remoteProcessPicker does not work
When F5'ing a launch.json that is configured to use remoteProcessPicker on windows, it will fail to send the script correctly and return a `....` command not found.

### PR DESCRIPTION
**Pull request by WardenGnaw**
The windows remote attach to process picker was not working due to double quotes then single quotes on the host machine. 

This PR fixes this issue and selects the correct quotes depending on the correct host machine. Fixes https://github.com/microsoft/vscode-cpptools/issues/3758

| Host | Remote | Status|
| ------------- |:-------------:| -----:|
| Windows     | Linux | ✔ |
| Windows     | macOS | ✔ |
| macOS     | Linux | ✔ |
| Linux | macOS| ✔ |

### CODE REVIEW

**Comment by sean-mcmanus** [comment]
Was a GitHub issue filed on this?

**Review by sean-mcmanus — APPROVED** [review]

**Comment by WardenGnaw** [comment]
> Was a GitHub issue filed on this?

Not that I know of. I ran across this while doing other work.

**Comment by pieandcakes** [comment]
@WardenGnaw you should create an issue and link it. people who run into problems search issues and this would help explain the problem while being findable on the issues search.

File: Extension/src/Debugger/attachToProcess.ts [hunk 1]
```diff
@@ -103,15 +103,29 @@ export class RemoteAttachPicker {
         });
     }
 
+    // Creates a string to run on the host machine which will execute a shell script on the remote machine to retrieve OS and processes
+    private getRemoteProcessCommand(): string {
+        let shellCommandQuote: string = `'`;
+        let hostCommandQuote: string = `"`;
+
+        // Must use single quotes around the whole command and double quotes for the argument to `sh -c` because Linux evaluates $() inside of double quotes.
```
Comment (pieandcakes) [inline Extension/src/Debugger/attachToProcess.ts:111]: Can you call it outer and inner quote? isn't the whole thing also a shell command?

File: Extension/src/Debugger/attachToProcess.ts [hunk 2]
```diff
@@ -103,15 +103,29 @@ export class RemoteAttachPicker {
...
+
+        return `${hostCommandQuote}sh -c ${shellCommandQuote}uname && if [ $(uname) = \\\"Linux\\\" ] ; then ${PsProcessParser.psLinuxCommand} ; elif [ $(uname) = \\\"Darwin\\\" ] ; ` +
+        `then ${PsProcessParser.psDarwinCommand}; fi${shellCommandQuote}${hostCommandQuote}`;
+    }
+
     private getRemoteOSAndProcesses(pipeCmd: string): Promise<AttachItem[]> {
-        // Commands to get OS and processes
-        const command: string = `sh -c "uname && if [ $(uname) = \\\"Linux\\\" ] ; then ${PsProcessParser.psLinuxCommand} ; elif [ $(uname) = \\\"Darwin\\\" ] ; ` +
-            `then ${PsProcessParser.psDarwinCommand}; fi"`;
+        // Do not put any quoting in execCommand.
```
Comment (pieandcakes) [inline Extension/src/Debugger/attachToProcess.ts]: `do not "add"` instead of "put" ?

**Comment by pieandcakes** [comment]
also please add your test matrix to this in case we run into a problem in the future.

**Review by pieandcakes — APPROVED** [review]

**Pull request merged**